WikiWatershed/model-my-watershed
# Implementing logging in

Implements the login modal.

To test:
- Click sign in filling in any of the fields. You should see error messages for both fields.
- Click sign in with only the username field filled in. You should see an error message for the password field.
- Click sign in with only the password field filled in. You should see an error message for the username.
- Enter junk in both fields. You should see an error message about the login and password.
- Enter good credentials (create a user with `./scripts/manage createsuperuser` if you haven't already). The modal should dismiss when the response comes back, and the header should update `ctaylor` to whatever username you entered. 

Some issues as a result of this:
- https://github.com/azavea/model-my-watershed/issues/149 - We now need a view to manage the header.
- https://github.com/azavea/model-my-watershed/issues/150 - We need a way to reopen the modal. Probably should be done as part of 149.
- https://github.com/azavea/model-my-watershed/issues/151 - There are a few states that aren't currently in the mock-ups.
- https://github.com/azavea/model-my-watershed/issues/152 - Implement sign-up and forgot username/password.
- https://github.com/azavea/model-my-watershed/issues/153 - Add tests for the both the back-end and front-end user app. 
- https://github.com/azavea/model-my-watershed/issues/155 - Do want to use the Django registration templates?

Connects to #129
